cumulus/util/tropo.py

```diff
@@ -22,8 +22,9 @@ def get_resource_by_type(template, type_to_find):
             if item.__class__ is type_to_find:
                 result.append(item)
 
-        if len(result) == 0:
-            raise ValueError("Expected to find resource of type %s in template but did not." % type_to_find)
+        # TODO: review: how does this affect other behaviours..
```

Comment:
Might want to do an empty check in `get_pipeline_stage_by_name()` now, so that we could still raise this error, if consumer code attempts to attach to a pipeline stage that doesn't exist.

Reply:
This comment caught me off guard a bit because it's around pipelines and this PR is params!   However, when implementing I stumbled on some code I couldn't really work around.   

That method already has an empty check though.  
https://github.com/brettswift/cumulus/blob/master/cumulus/util/tropo.py#L50

Reply:
oh, you quoted "by_name" not "by_type" lol

Reply:
oh nevermind, i see what you mean now

Reply:
So,  to clarify this fix isn't in case someone attaches a stage that doesn't exist,the modification here raises an error if there is no pipeline in the template. 

The detection of not finding a stage when expecting one is already in the code base.